tests/Test.Yaapii.Http/Parts/Bodies/BodyTests.cs

```diff
@@ -21,23 +21,75 @@
 //SOFTWARE.
 
 using Xunit;
+using Yaapii.Atoms.IO;
+using Yaapii.Atoms.Text;
 using Yaapii.Http.AtomsTemp.Lookup;
+using Yaapii.Http.Fake;
+using Yaapii.Http.Mock;
+using Yaapii.Http.Requests;
+using Yaapii.Http.Responses;
+using Yaapii.Http.Wires;
+using Yaapii.Http.Wires.AspNetCore;
 
 namespace Yaapii.Http.Parts.Bodies.Test
 {
+    [Collection("actual http tests")]
     public sealed class BodyTests
     {
         [Fact]
         public void WritesBody()
         {
             Assert.Equal(
                 "| <-- stick figure body",
-                new Body.Of(
-                    new Body("| <-- stick figure body").Apply(
-                        new Map.Of(new MapInput.Of())
+                new Base64Text(
+                    new TextBody.Of(
+                        new Body(
+                            new InputOf("| <-- stick figure body")
+                        ).Apply(
+                            new Map.Of(new MapInput.Of())
+                        )
                     )
                 ).AsString()
             );
         }
+
+        [Fact]
+        public void TransmitsZipFile()
+        {
+            var port = new AwaitedPort(new RandomPort().Value()).Value();
+            using (var server =
+                new HttpMock(port,
+                    new FkWire(req =>
+                        new Response.Of(
+                            new Status(200),
+                            new Reason("OK"),
+                            new Body(
+                                new ResourceOf(
+                                    "Assets/test.zip",
```

Comment:
What you are testing here is, that something, which is encoded by ```new Body(...)``` is the same after going through the wire and be decoded by ```new Body.Of(...)```. It does not answer the question: "Can I send a raw zip file to a client (which propably does not use Yaapii.Http to process the request) without needing him to use Base64 decoding.

The main problem here is the usage of ```HttpMock``` which already translates the request.InputStream to a Dictionary<string, string>. To make it more clear you could directly use the ```MockServer``` class, which represents your client, which you want to send the zip-file to:
```csharp
 [Fact]
        public void TransmitsRawZipFile()
        {
            var port = new AwaitedPort(new RandomPort().Value()).Value();
            IInput result = new DeadInput();
            using (var server =
                new MockServer(
                    port,
                    "{}",
                    (req, res, prm) => 
                        result = 
                            new InputOf(
                                new BytesOf(
                                    new InputOf(req.InputStream)
                                ).AsBytes()
                            ),
                    "localhost"
                )
            )
            {
                new AspNetCoreWire(
                    new AspNetCoreClients()
                ).Response(
                    new Post($"http://localhost:{port}/",
                        new Body(
                            new ResourceOf(
                                "Assets/test.zip",
                                this.GetType().Assembly
                            )
                        )
                    )
                );
            }
            Assert.Equal(
                "this is a test", // content of test.txt in Assets/test.zip
                new TextOf(
                    new UnzippedFile(result, "test.txt")
                ).AsString()
            );

        }
```
The AspNetCoreWire posts a Zip-File and the request InputStream is expected to be a zip File. You will see, that the resulting zip is unreadable, because the bytes are Base64 encoded.